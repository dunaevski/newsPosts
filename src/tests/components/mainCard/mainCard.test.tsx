import React from 'react';
import {render, screen} from '@testing-library/react';
import MainCard from '../../../components/mainCard/mainCard';

it('test MainCard snap', () => {
    const props = {
        title: 'Title',
        content: true,
    }

    const {asFragment} = render(<MainCard {...props}>Children</MainCard>);
    const firstRender = asFragment()

    expect(firstRender).toMatchSnapshot()

    expect(screen.getByText('Title')).toBeInTheDocument();
    expect(screen.getByText('Children')).toBeInTheDocument();
});
