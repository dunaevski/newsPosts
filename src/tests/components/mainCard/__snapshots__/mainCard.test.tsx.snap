// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders welcome message 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vnsqsq-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-vxmz7y-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          Title
        </span>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      Children
    </div>
  </div>
</DocumentFragment>
`;
